### PR TITLE
Cherry-picking changes of RRFS_dev1 between Sep 7 and Oct 19

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 

- Cherry-pick the change of the GSL RRFS_dev1 workflow between 09/07/2023 - 10/19/2023.
- Update module files based on the GSL RRFS_dev1 branch of the UFS SRW App.
- Fix the CRTM version issue on WCOSS2.
- Update the readme and quick start guide.

## TESTS CONDUCTED: 

- On machines/platforms:

- [x] WCOSS2
- [x] Hera
- [ ] Orion
- [ ] Jet

- Test cases: 

- [ ] Non-DA engineering test
- [x] DA engineering test
- [ ] Other sample scripts:

## ISSUE: 

- Fixes the issue(s) mentioned in #49

### CODE REVIEW

**Comment by chan-hoo** [comment]
@hu5970, the DA sample run was completed successfully on Hera. However, some tasks using `update_ice`, `update_GVF`, and `adjust_soiltq` failed with the following message:
```
 >>> open file: sfc_data.nc
 get_var_nc_realERROR:      1987440  should equal to     10665000

```
I think this is caused by the domain mismatch. In the DA sample runs, RRFS_CONUS_3km is used. What do you think of this?

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, when you merge this PR, please do not 'squash and merge' this.

**Comment by MatthewPyle-NOAA** [comment]
Awesome on reaching this point, @chan-hoo .  Will try to review it tomorrow, and will avoid a squash and merge when that time comes.

**Comment by hu5970** [comment]
> @hu5970, the DA sample run was completed successfully on Hera. However, some tasks using `update_ice`, `update_GVF`, and `adjust_soiltq` failed on WCOSS2 with the following message:
> 
> ```
>  >>> open file: sfc_data.nc
>  get_var_nc_realERROR:      1987440  should equal to     10665000
> ```
> 
> I think this is caused by the domain mismatch. In the DA sample runs, RRFS_CONUS_3km is used. What do you think of this?

I agree there are some mismatch in dimensions of those files. Could you send me the test location so I can take a look? If you turn off those steps for testing, that is OK. We can figure this out during following up test after this PR merged.

**Comment by BenjaminBlake-NOAA** [comment]
@chan-hoo I should also be able to review this PR tomorrow - thanks!

**Comment by chan-hoo** [comment]
@hu5970, 
- log file: 
```
/lfs/h2/emc/ptmp/chan-hoo.jeon/test_da/logs/rrfs.20231103/04/prep_cyc_spinup_c0v00_2023110304.log
```
- stdout_cycleICE file:
```
/lfs/h2/emc/stmp/chan-hoo.jeon/test_da/2023110304/fcst_fv3lam_spinup/INPUT/stdout_cycleICE
```

**Review by MatthewPyle-NOAA — COMMENTED** [review]
Really just had one comment about the modules in one of the task files, but otherwise thought it looked good.  Ran a successful nonDA engineering test on WCOSS2.

File: modulefiles/tasks/wcoss2/run_prepstart.local.lua
Comment (chan-hoo) [reply modulefiles/tasks/wcoss2/run_prepstart.local.lua:1]: @MatthewPyle-NOAA, yes, that is my concern, too. However, @hu5970 mentioned that the specific (old) versions should be used for this task. Since I think this doesn't meet the NCO standards, we should update this file in the future.

Comment (BenjaminBlake-NOAA) [reply modulefiles/tasks/wcoss2/run_prepstart.local.lua:1]: @MatthewPyle-NOAA @chan-hoo  This is for the FVCOM processing, which is part of the run_prepstart task, because the newer version of the nco module does not work properly.

**Review by MatthewPyle-NOAA — DISMISSED** [review]
Approving, but will wait on Ben before merging.

File: scripts/exrrfs_run_postanal.sh
Comment (BenjaminBlake-NOAA) [inline scripts/exrrfs_run_postanal.sh:166]: Found a typo - aerosol

**Review by BenjaminBlake-NOAA — APPROVED** [review]
@chan-hoo  I found one typo but otherwise the changes look good to me!  I was going to run a non-DA test on WCOSS2, but since Matt already did I don't want to hold up the merging of this PR.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Reapproving updated version

**Comment by MatthewPyle-NOAA** [comment]
Are we ready for a non-squashed merge at this point?

**Comment by chan-hoo** [comment]
yes, I think so.

**Pull request merged**
